### PR TITLE
ci: upgrade flutter from 2.2.1 to 2.5.2 (blocked by #491)

### PR DESCRIPTION
**Pull request by juliansteenbakker**
## Description

This will upgrade the CI flutter version to 2.5.2.

## Related Issues

This will depend on #466 since it will not pass with the deprecated integration_test.
This is also blocking #442.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ X] All existing and new tests are passing.
- [X ] I updated/added relevant documentation (doc comments with `///`).
- [X ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ X] I read and followed the [Flutter Style Guide].
- [ X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by juliansteenbakker** [comment]
This is blocked by #466

**Comment by juliansteenbakker** [comment]
It seems like some tests are failing with latest flutter release. I'll check them and update this PR ASAP.

**Comment by mhadaily** [comment]
thanks, I'll review again tomorrow morning.

File: packages/package_info_plus/package_info_plus_windows/lib/src/file_version_info.dart [hunk 1]
```diff
@@ -10,10 +10,10 @@ part of package_info_plus_windows;
 
 class _LANGANDCODEPAGE extends Struct {
   @Uint16()
-  int? wLanguage;
+  external int? wLanguage;
 
   @Uint16()
-  int? wCodePage;
+  external int? wCodePage;
 }
```
Comment (mhadaily) [inline packages/package_info_plus/package_info_plus_windows/lib/src/file_version_info.dart:17]: This PR is mentioning CI but we have a change in the package. can you take this out and send another PR for this. Please mention what this change is going to resolve.

File: packages/package_info_plus/package_info_plus_windows/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: package_info_plus_windows
 description: Windows implementation of the package_info_plus plugin
-version: 1.0.3
+version: 1.0.4
```
Comment (mhadaily) [inline packages/package_info_plus/package_info_plus_windows/pubspec.yaml:3]: see above

File: packages/package_info_plus/package_info_plus_windows/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 1.0.4
+- Annotate int with external.
+
```
Comment (mhadaily) [inline packages/package_info_plus/package_info_plus_windows/CHANGELOG.md:3]: see my comment

**Review by mhadaily — CHANGES_REQUESTED** [review]
please consider my comments

File: packages/package_info_plus/package_info_plus_windows/CHANGELOG.md (hunk 3 above)
Comment (juliansteenbakker) [reply packages/package_info_plus/package_info_plus_windows/CHANGELOG.md]: Fixed, please see #491

**Comment by juliansteenbakker** [comment]
I have moved the required changes to the windows package into #491

**Comment by mhadaily** [comment]
thanks, I have prepared a new PR #489 that covers this. I'll close this PR but will consider your contribution for sure. you have done a great job. thanks

**Pull request closed without merging**